app/helpers/application_helper.rb

```diff
@@ -620,6 +620,25 @@ def user_specific_javascript_includes
     tags.html_safe
   end
 
+  # Properly decides main menu expanded state before its drawn
+  # Fixes flickering side menu where menu is first expanded, then being
+  # collapsed in angular.
+  def expand_menu_javascript
+    javascript_tag(%{
+      var wrapper = jQuery('#wrapper');
+      if (window.sessionStorage.getItem('openproject:navigation-toggle') === 'collapsed') {
+        jQuery('#wrapper,#breadcrumb,#content,#toggle-project-menu').addClass('hidden-navigation');
+      }
+      wrapper.show();
+    })
```

Comment:
As discussed, I'd like for that code to not be inline but rather where the rest of the menu toggle code is placed.